doc: Added Thread 1.2 features and configuration documentation. (#2712)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_overview.rst

* All device roles
* Border Router support

To stay up to date with the newest Thread trends, the |NCS| also gradually implements features introduced in Thread 1.2 version.

> This protocol version is fully backward compatible with Thread 1.1 (including Thread 1.1.1) and improves networks E.g. by making it more scalable, capable, responsive and low-power.

Reviewer: ```suggestion
This protocol version is fully backward compatible with Thread 1.1 (specifically Thread 1.1.1) and improves networks e.g. by making them more scalable, capable, responsive and low-power.
```

For more information about Thread 1.2 see official materials at `Thread Group's Thread 1.2 Commercial White Paper`_ and `Thread Group's Thread 1.2 Base Features`_.

You may choose the version of Thread protocol, which will be used in your application.
By default |NCS| supports Thread 1.1, but you can enable and configure Thread 1.2 using dedicated options (see :ref:`thread_ug_thread_1_2` for details).

> |NCS| does not support all Thread 1.2 functionalities yet, so under above link you can also find the list of currently available features with information on how to enable it.

Reviewer: I would mention about the currently available functionalities in the configuration section.

* All device roles
* Border Router support

To stay up to date with the newest Thread trends, the |NCS| also gradually implements features introduced in Thread 1.2 version.
This protocol version is fully backward compatible with Thread 1.1.1 and improves networks E.g. by making it more scalable, capable, responsive and low-power.

For more information about Thread 1.2 see official materials at `Thread Group's Thread 1.2 Commercial White Paper`_ and `Thread Group's Thread 1.2 Base Features`_.

> |NCS| does not support all Thread 1.2 functionalities yet, so below you can see the list of currently available features:

Reviewer: Should we move this list to the `thread_ug_thread_1_2`? So that there is only one place with the detailed Thread 1.2 information and it's easier to maintain.

Author: If it comes to maintenance you are right, but do you think that it would be ok from the user point of view? My intention was to not suggest in overview that we support only Thread 1.1.1 features and if they are listed there, to also list Thread 1.2 features. Moreover thread_ug_thread_1_2 is located in Configuring Thread in NCS, which is mainly concentrated on configuring and I think that in some other placed it should be mentioned that user may found something like Thread 1.2 options. @greg-fer what do you think about this?

Reviewer: My thinking is that at the moment it's okay to have just a three items list, but when more Thread 1.2 functionalities come (https://github.com/openthread/openthread/issues/4327) there will be a need to explain how to enable/disable them (in the configuration section) and that would be redundant with this section.

Reviewer 2: I'm a bit confused, to be honest. I think we need a clear separation about what is the default setting, and what are the extra version you can enable. Concerning the Kconfig options list, IF the v1.2 options will be all optional, and v1.1.1 options are enabled by default, I think we can mention the v1.2 options only on the configuration page.

Author: I think that @edmont is right and it will be better to make overview page general and place whole detail information in the configuration section. In the future it will be easier to expand it and even evolve it to Thread version options, not only Thread 1.2 options (if next Thread versions will be released).
